pkg/cache/groupcache.go

```diff
@@ -154,7 +162,11 @@ func NewGroupcacheWithConfig(logger log.Logger, reg prometheus.Registerer, conf
 				if err != nil {
 					return err
 				}
-				return dest.UnmarshalBinary(encodedList)
+
+				if isBlocksRootDir(parsedData.Name) {
+					return dest.UnmarshalBinary(encodedList, time.Now().Add(BlocksIterTTL))
+				}
+				panic("caching bucket layer must not call on unconfigured paths")
```

Comment:
I was under the impression we should never use panic. Shouldn't we be returning an error instead of panicking?

Reply:
I think that using `panic`s to check for logic bugs in code is one of the very few valid use cases for it :smile: